NineAnimator/Models/Anime Source/animeworld.tv/AnimeWorld+Search.swift

```diff
@@ -0,0 +1,104 @@
+                requestTask = parent.request(
+                    browsePath: "/search",
+                    query: [ "keyword": title ]
+                ) .then {
+                    [weak self] response -> [AnimeLink] in
+                    guard let self = self else { throw NineAnimatorError.unknownError }
+                    let bowl = try SwiftSoup.parse(response)
+                    let entries = try bowl.select("div.film-list>div.item")
+                    let resultingLinks = entries.compactMap {
+                        entry -> AnimeLink? in
+                        do {
+                            let animeTitle = try entry.select("a.name"/*"h6.card-title>b"*/)
+                            let title = try animeTitle.text()
+                            if title.isEmpty { return nil }
+                            let animeLinkPath = try entry.select("div>a")
+                            let url = try animeLinkPath.attr("href")
+                            let animeUrl = try URL(
+                                string: url,
+                                relativeTo: self.parent.endpointURL
+                            ).tryUnwrap()
+                            let animeArtworkPath = try entry.select("img").attr("src")
+                            
+                            guard let coverImage = URL(string: animeArtworkPath)
+                            else {
+                                return nil
+                            }
+                            return AnimeLink(
+                                title: title,
+                                link: animeUrl,
+                                image: coverImage,
+                                source: self.parent
+                            )
+                        } catch { return nil }
+                    }
+                    
+                    guard !resultingLinks.isEmpty else {
+                        throw NineAnimatorError.searchError("No results found")
+                    }
+                    
+                    return resultingLinks
+                } .error {
+                    [weak self] error in
+                    guard let self = self else { return }
+                    self.delegate?.onError(error, from: self)
+                    self.requestTask = nil
+                } .finally {
+                    [weak self] results in
+                    guard let self = self else { return }
+                    self._results = results
+                    self.delegate?.pageIncoming(0, from: self)
+                    self.requestTask = nil
+                }
+            }
+            
+            init(query: String, parent: NASourceAnimeWorld) {
+                self.title = query
+                self.parent = parent
+            }
+        }
+        
+        func search(keyword: String) -> ContentProvider {
+            SearchAgent(query: keyword, parent: self)
+        }
```

Comment:
of course!

Reply:
I have not understand how I can fix the `Add files via upload` in this pull, I have did all different commit because using web interface I have not found a way to commit one time. I have downloaded GitHub desktop and now I can do it

Reply:
If you are using Xcode on a macOS computer, then you should be able to clone and push to your fork of NineAnimator.

Reply:
ok, so I have to clone your master, open it with Xcode, add anime world file and push update to my repository?